pkg/agent/watcher.go

```diff
@@ -104,86 +95,155 @@ func (w *watcher) ConfigChan() <-chan resmgrConfig {
 
 // GetConfig returns the current cri-resmgr configuration
 func (w *watcher) GetConfig() resmgrConfig {
-	return w.currentConfig.get()
+	cfg, kind := w.currentConfig.get()
+	w.Info("giving %s configuration in reply to query", kind)
+	return cfg
+}
+
+// sendConfig sends the current configuration.
+func (w *watcher) sendConfig() {
+	cfg, kind := w.currentConfig.get()
+	w.Info("pushing %s configuration to client", kind)
+	w.configChan <- cfg
 }
 
 func (w *watcher) watch() error {
-	var cmw watch.Interface
-	// Start with a closed channel. We want this so that we end up in creating
-	// a new watch in the main event loop below
-	eventChan := func() <-chan watch.Event {
-		c := make(chan watch.Event)
-		close(c)
-		return c
-	}()
+	nodew := newNodeWatch(w)
+	group := ""
+
+	if node, err := nodew.Query(); err != nil {
+		w.Warn("failed to query node %s: %v", nodeName, err)
+	} else {
+		group = node.(*core_v1.Node).Labels[opts.labelName]
+		w.Info("configuration group is set to '%s'", group)
+	}
+
+	cfgw := newConfigMapWatch(w, opts.configMapName+".node."+nodeName, namespace(opts.configNs))
+	grpw := newConfigMapWatch(w, groupMapName(group), namespace(opts.configNs))
```

Comment:
Hmm, shouldn't we add a watch for the default map as well? If I read this correctly the implemented logic is not quite as the commit message suggests (which I think makes sense).

I think that in the case that the node has group label set but the group map does not exists, we should still return the default map(?) Or what do you think?

Reply:
Well, we track the 2 necessary of the 3 ConfigMaps. We either track the group or the default one, but never both simultaneously.

Initially we used to track all 3 ConfigMap simultaneously and the logic for missing group configuration was what you suggest now. I deliberately changed it to be what it is now. I don't have a strong opinion about this, but it felt more natural to have it this way.